src/main/kotlin/app/CacheService.kt

```diff
@@ -0,0 +1,84 @@
+@file:Suppress("SqlResolve")
+
+package app
+
+import app.util.HikariCpDataSource
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import org.slf4j.LoggerFactory
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+
+object CacheService {
+    private val log = LoggerFactory.getLogger(CacheService.javaClass)
+    private val objectMapper = jacksonObjectMapper()
+
+    private fun createTableIfAbsent() {
+        val connection = HikariCpDataSource.connection
+        val statement = connection.createStatement()
+
+        statement.execute(
+            "CREATE TABLE IF NOT EXISTS userinfo (" +
+            "id VARCHAR2 PRIMARY KEY, " +
+            "timestamp TIMESTAMP, " +
+            "data JSON" +
+            ")"
+        )
+    }
+
+    fun selectJsonFromDb(username: String): String? {
+        val connection = HikariCpDataSource.connection
+
+        createTableIfAbsent()
```

Comment:
Switch these two? The `createTableIfAbsent()` method creates its own connection.

Reply:
Oops, missed this one. It's also resolved by [the below](https://github.com/tipsy/profile-summary-for-github/pull/113/commits/e9abd1979813e1c78dad544734dedc335df6a933#diff-50edd97fe3c397a256ba5e035a23f92fbdeac2dffbf60e80baa32021e2b43577L31-L32).